Unexpected behavior with whitespace in comma separated elements

I've noticed that pedantic-pom-enforcers preserves whitespace in comma-separated values. This can lead to unexpected behavior - for example:

```xml
<compound implementation="com.github.ferstl.maven.pomenforcers.CompoundPedanticEnforcer"> 
 
 <enforcers>POM_SECTION_ORDER,MODULE_ORDER,DEPENDENCY_MANAGEMENT_ORDER,DEPENDENCY_ORDER,DEPENDENCY_CONFIGURATION,DEPENDENCY_ELEMENT,DEPENDENCY_SCOPE,PLUGIN_MANAGEMENT_ORDER,PLUGIN_CONFIGURATION</enforcers>
  <!-- DEPENDENCY_SCOPE configuration -->
  <testDependencies>
    org.assertj:assertj-core,
    org.junit.jupiter:junit-jupiter,
    org.junit.jupiter:junit-jupiter-api,
    org.junit.jupiter:junit-jupiter-engine,
    org.junit.jupiter:junit-jupiter-params
  </testDependencies>
</compound>
```

This gets parsed internally into ArtifactModel keys:

```
\n                                        org.junit.jupiter:junit-jupiter-engine:
org.assertj:assertj-core:
\n                                        org.junit.jupiter:junit-jupiter-params:
\n                                        org.junit.jupiter:junit-jupiter:
\n                                        org.junit.jupiter:junit-jupiter-api:
```

As a result, the test dependencies aren't validated unless they are the first one listed.

You can also reproduce this with the list of enforcers:

```xml
<enforcers>
  POM_SECTION_ORDER,
  MODULE_ORDER,
  DEPENDENCY_MANAGEMENT_ORDER,
  DEPENDENCY_ORDER,
  DEPENDENCY_CONFIGURATION,
  DEPENDENCY_ELEMENT,
  DEPENDENCY_SCOPE,
  PLUGIN_MANAGEMENT_ORDER,
  PLUGIN_CONFIGURATION
</enforcers>
```

which results in the error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce) on project ...: Unable to parse configuration of mojo org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce for parameter enforcers: Cannot set 'enforcers' in class com.github.ferstl.maven.pomenforcers.CompoundPedanticEnforcer: InvocationTargetException: No enum constant com.github.ferstl.maven.pomenforcers.PedanticEnforcerRule.
[ERROR]                                         MODULE_ORDER
```

To fix, I think adding a `.map(String::trim)` before this line will allow whitespace in comma separated values: https://github.com/ferstl/pedantic-pom-enforcers/blob/9b1ef0b1c83f5b0b07afdfb3c3778e8ea3d7f6dc/src/main/java/com/github/ferstl/maven/pomenforcers/util/CommaSeparatorUtils.java#L39

--- a/src/main/java/com/github/ferstl/maven/pomenforcers/util/CommaSeparatorUtils.java
+++ b/src/main/java/com/github/ferstl/maven/pomenforcers/util/CommaSeparatorUtils.java
@@ -23,7 +23,7 @@ import static java.util.stream.Collectors.toCollection;
 
 public final class CommaSeparatorUtils {
 
-  private static final Splitter COMMA_SPLITTER = Splitter.on(",");
+  private static final Splitter COMMA_SPLITTER = Splitter.on(",").trimResults().omitEmptyStrings();
 
   public static void splitAndAddToCollection(String commaSeparatedItems, Collection<String> collection) {
     splitAndAddToCollection(commaSeparatedItems, collection, Function.identity());

--- a/src/test/java/com/github/ferstl/maven/pomenforcers/util/CommaSeparatorUtilsTest.java
+++ b/src/test/java/com/github/ferstl/maven/pomenforcers/util/CommaSeparatorUtilsTest.java
@@ -1,0 +1,28 @@
+package com.github.ferstl.maven.pomenforcers.util;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class CommaSeparatorUtilsTest {
+    @Test
+    public void testSplitAndAddToCollection() {
+        final Map<String, List<String>> tests = ImmutableMap.<String, List<String>> builder()
+            .put("a,b,c", ImmutableList.of("a", "b", "c"))
+            .put("a,b,,,c,", ImmutableList.of("a", "b", "c"))
+            .put(" a \n,\n   \t b   ,\r\n  c \t\n", ImmutableList.of("a", "b", "c"))
+            .build();
+        for (Map.Entry<String, List<String>> test : tests.entrySet()) {
+            final List<String> l = new ArrayList<>();
+            CommaSeparatorUtils.splitAndAddToCollection(test.getKey(), l);
+            assertThat(l, equalTo(test.getValue()));
+        }
+    }
+}

